### PR TITLE
etcd-encryption: update for backup/restore

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md [hunk 1]
```diff
@@ -270,6 +270,30 @@ openshift-apiserver:
...
+` and mounted via host mount as `/etc/kubernetes/static-pod-resources/secrets/encryption-config/encryption-config` in the kube-apiserver pod.
+
+A restore must put both parts of the backup in place in master file-system before starting up kube-apiserver. 
+
+Note: the restore of the openshift-apiserver encryption config will be automatic if both (1) and (2) match.
+
+The backup of (2) must happen shortly after (1) (before another write-key is set, which happens once per week): assume (1) happens
+
+a. outside of migration: then (2) can happen
+   - before migration too
```
Comment (p0lyn0mial) [inline enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md]: outside but before? not sure if I understand the difference between `a` and `b`

Comment (sttts) [reply enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md]: a) 
* `1--2--(migration)`
* `1--(-2-migration)`
* `1--(migration)-2--`
b) 
* `(--1--2--migration)`
* `(--1--migration)-2--`

File: enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md [hunk 2]
```diff
@@ -270,6 +270,30 @@ openshift-apiserver:
...
+   - during a just started migration
+   - after a started migration.
+   In the first case, config and etcd snapshot perfectly match. In the second and third case, a new write-key is part of the config and the last read-key is preserved in the config as well allowing to read newly written data and untouched data.
```
[the remark this answers is not in the document]
Comment (sttts) [reply enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md]: fixed

File: enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md [hunk 3]
```diff
@@ -270,6 +270,30 @@ openshift-apiserver:
...
+
+A backup consists of the steps 
+
+1. backup the etcd snapshot
+2. backup the master file `/etc/kubernetes/static-pod-resources/kube-apiserver-pod-<REVISION>/secrets/encryption-config/encryption-config
+` and mounted via host mount as `/etc/kubernetes/static-pod-resources/secrets/encryption-config/encryption-config` in the kube-apiserver pod.
+
+A restore must put both parts of the backup in place in master file-system before starting up kube-apiserver. 
+
+Note: the restore of the openshift-apiserver encryption config will be automatic if both (1) and (2) match.
```
[the remark this answers is not in the document]
Comment (sttts) [reply enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md]: Why? openshift-apiserver is not manually restored (to be tested by QA, cc @xingxingxia)

File: enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md (hunk 1 above)
Comment (p0lyn0mial) [reply enhancements/kube-apiserver/encrypting-data-at-datastore-layer.md]: much better, thx :)

**Pull request merged**